contracts/governance/Governor.sol

```diff
@@ -4,200 +4,233 @@ pragma solidity ^0.8.0;
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        string memory description
+    ) public virtual override returns (uint256) {
+        uint256 proposalId = hashProposal(targets, values, calldatas, keccak256(bytes(description)));
+
+        require(targets.length == values.length, "Governor: invalid proposal length");
+        require(targets.length == calldatas.length, "Governor: invalid proposal length");
+        require(targets.length > 0, "Governor: empty proposal");
+
+        ProposalCore storage proposal = _proposals[proposalId];
+        require(proposal.voteStart.isUnset(), "Governor: proposal already exists");
+
+        uint64 snapshot = uint64(block.number) + votingDelay();
+        uint64 deadline = snapshot + votingPeriod();
+
+        proposal.voteStart.setDeadline(snapshot);
+        proposal.voteEnd.setDeadline(deadline);
+
+        emit ProposalCreated(
+            proposalId,
+            _msgSender(),
+            targets,
+            values,
+            new string[](targets.length),
+            calldatas,
+            snapshot,
+            deadline,
+            description
         );
-        return _castVote(proposalId, voter, support);
+
+        return proposalId;
     }
 
     /**
-     * @dev Internal propose mechanism: Hashes proposal and sets snapshot and deadline. Revert if proposal is already
-     * registered.
-     *
-     * Note: does not emit any events. Events are part of the public function so they can be customized.
+     * @notice module:core
+     * @dev See {IGovernor-execute}.
      */
-    function _propose(
+    function execute(
         address[] memory targets,
         uint256[] memory values,
         bytes[] memory calldatas,
-        bytes32 salt
-    )
-        internal virtual returns (uint256 proposalId, uint256 snapshot, uint256 deadline)
-    {
-        proposalId = hashProposal(targets, values, calldatas, salt);
+        bytes32 descriptionHash
+    ) public payable virtual override returns (uint256) {
+        uint256 proposalId = hashProposal(targets, values, calldatas, descriptionHash);
 
-        require(targets.length == values.length,    "Governance: invalid proposal length");
-        require(targets.length == calldatas.length, "Governance: invalid proposal length");
-        require(targets.length > 0,                 "Governance: empty proposal");
+        require(state(proposalId) == ProposalState.Succeeded, "Governor: proposal not successfull");
+        _proposals[proposalId].executed = true;
 
-        Proposal storage proposal = _proposals[proposalId];
-        require(proposal.timer.isUnset(), "Governance: proposal already exists");
+        for (uint256 i = 0; i < targets.length; ++i) {
+            if (calldatas[i].length == 0) {
+                Address.sendValue(payable(targets[i]), values[i]);
+            } else {
+                Address.functionCallWithValue(targets[i], calldatas[i], values[i]);
+            }
```

Comment:
Re-raising a previous comment in case it gets lost.

I would like us to use `.call`. I think we agreed to make `Address._verifyCallResult` an internal function instead of a private helper, so we can reuse it here.